src/Fantomas/SourceParser.fs

```diff
@@ -1761,8 +1761,9 @@ let (|ElmishReactWithChildren|_|) (e: SynExpr) =
     | SynExpr.App (_,
                    false,
                    SynExpr.App (_, false, OptVar ident, (ArrayOrList _ as attributes), _),
-                   ArrayOrList (sr, isArray, children, er, _),
-                   _) -> Some(ident, attributes, (isArray, sr, children, er))
+                   ArrayOrList (sr, isArray, children, er, r),
+                   _) when (not (RangeHelpers.isAdjacentTo attributes.Range r)) ->
```

Comment:
Hey David,

This fix looks fine but perhaps we can use the same fix here as `ElmishReactWithoutChildren`, where we just say upfront that `IndexWithoutDotExpr` does not count.

But what I'm also wondering is if we need to combine these three active patterns into one as they handle the same AST shape.

Something like:
```fsharp
let (|IndexWithoutDotExpr|ElmishReactWithoutChildren|ElmishReactWithChildren|NonAppExpr|) = ...
```
and just never do anything with `NonAppExpr`.

Would you be interested in exploring combining the patterns?

Reply:
Yeah, sure. I'll try that in the next couple of days.

Reply:
Something like this?
Gotta say, I really like your idea here.